threedi_modelchecker/config.py

```diff
@@ -1176,221 +1142,215 @@ def is_none_or_empty(col):
-            ~is_none_or_empty(models.GlobalSetting.interception_file),
-            is_none_or_empty(models.GlobalSetting.interception_global),
+        column=models.Interception.interception,
+        invalid=Query(models.Interception).filter(
+            ~is_none_or_empty(models.Interception.interception_file),
+            is_none_or_empty(models.Interception.interception),
         ),
-        message="v2_global_settings.interception_global is recommended as fallback value when using an interception_file.",
+        message="interception.interception is recommended as fallback value when using an interception_file.",
     ),
 ]
 
 CHECKS += [
-    QueryCheck(
-        error_code=326,
-        level=CheckLevel.INFO,
-        column=table.id,
-        invalid=Query(table).filter(
-            table.id != Query(setting).filter(first_setting_filter).scalar_subquery()
-        ),
-        message=f"{table.__tablename__} is defined, but not referred to in v2_global_settings.{setting.name}",
+    UsedSettingsPresentCheck(
+        error_code=326, level=CheckLevel.ERROR, column=use_col, settings_table=table
     )
-    for table, setting in (
+    for table, use_col in (
         (
             models.SimpleInfiltration,
-            models.GlobalSetting.simple_infiltration_settings_id,
+            models.ModelSettings.use_simple_infiltration,
         ),
-        (models.Interflow, models.GlobalSetting.interflow_settings_id),
-        (models.GroundWater, models.GlobalSetting.groundwater_settings_id),
-        (models.NumericalSettings, models.GlobalSetting.numerical_settings_id),
-        (models.ControlGroup, models.GlobalSetting.control_group_id),
-        (models.VegetationDrag, models.GlobalSetting.vegetation_drag_settings_id),
+        (models.Interflow, models.ModelSettings.use_interflow),
+        (models.GroundWater, models.ModelSettings.use_groundwater_flow),
+        (models.GroundWater, models.ModelSettings.use_groundwater_storage),
+        (models.VegetationDrag, models.ModelSettings.use_vegetation_drag_2d),
+        (models.Interception, models.ModelSettings.use_interception),
     )
 ]
 
+# The test test_individual_checks prohibits setting min_length to 1 because then tests fail on empty dp
+CHECKS += [
+    SettingsLengthCheck(
+        error_code=326,
+        level=CheckLevel.INFO,
+        column=table.id,
+        min_length=min_length,
+        max_length=max_length,
+    )
+    for table, min_length, max_length in [
+        (models.ModelSettings, 0, 1),
+        (models.SimulationTemplateSettings, 0, 1),
+        (models.TimeStepSettings, 0, 1),
+        (models.NumericalSettings, 0, 1),
+        (models.PhysicalSettings, 0, 1),
+        (models.InitialConditions, 0, 1),
+        (models.SimpleInfiltration, 0, 1),
+        (models.Interflow, 0, 1),
+        (models.GroundWater, 0, 1),
+        (models.GroundWater, 0, 1),
+        (models.VegetationDrag, 0, 1),
+        (models.Interception, 0, 1),
+    ]
+]
+
+
```

Comment:
Check if settings tables do not have more than one row. Ideally, the check would confirm that `ModelsSettings`, `SimulationTemplateSettings`, `TimeStepSettings`, `NumericalSettings`, `PhysicalSettings` and `InitialConditions` have exactly one column, but that breaks some tests (see comment).

Reply:
See earlier remark, you should remove the min&max length parameters IMO

Reply:
Changed in 5ffa11850988199ed5f998476740e906b8837dea